boa/src/builtins/value/equality.rs

```diff
@@ -200,16 +200,16 @@ pub fn same_value_zero(x: &Value, y: &Value) -> bool {
 pub fn same_value_non_numeric(x: &Value, y: &Value) -> bool {
     debug_assert!(x.get_type() == y.get_type());
     match x.get_type() {
-        "undefined" => true,
-        "null" => true,
-        "string" => {
+        Type::Undefined => true,
+        Type::Null => true,
+        Type::String => {
             if x.to_string() == y.to_string() {
                 return true;
             }
             false
```

Comment:
This can be reduced further:
```suggestion
            x.to_string() == y.to_string()
```

Reply:
Fixed